flink/v2.1/flink/src/main/java/org/apache/iceberg/flink/sink/dynamic/CompareSchemasVisitor.java

```diff
@@ -88,7 +87,11 @@ public Result struct(Types.StructType struct, Integer tableSchemaId, List<Result
     }
 
     for (Types.NestedField tableField : tableSchemaType.asStructType().fields()) {
-      if (tableField.isRequired() && struct.field(tableField.name()) == null) {
+      if (tableField.isRequired()
+          && (caseSensitive
+                  ? struct.field(tableField.name())
+                  : struct.caseInsensitiveField(tableField.name()))
+              == null) {
```

Comment:
nit: Maybe create a method like `equalFieldNames` or something better named 😄, and then we hide the ugly details, and don't increase complexity?